driver/driver.c

```diff
@@ -130,6 +130,52 @@ WnbdDispatchPnp(PDEVICE_OBJECT DeviceObject,
         IoLocation->Parameters.DeviceCapabilities.Capabilities->Removable = 1;
         IoLocation->Parameters.DeviceCapabilities.Capabilities->EjectSupported = 1;
         break;
+    case IRP_MN_START_DEVICE:
+        {
+            if (NULL == GlobalExt || !GlobalExt->DeviceCount) {
+                break;
+            }
+            Status = WnbdGetScsiAddress(DeviceObject, &ScsiAddress);
+            if (Status) {
+                WNBD_LOG_ERROR("Could not query SCSI address. Error: %d.", Status);
+                break;
+            }
+
+            WNBD_LOG_INFO("Starting device.");
+            PWNBD_DISK_DEVICE Device = WnbdFindDeviceByAddr(
+                GlobalExt, ScsiAddress.PathId,
+                ScsiAddress.TargetId, ScsiAddress.Lun, TRUE);
+            if (!Device) {
+                break;
+            }
+
+            PDEVICE_OBJECT AttachedDisk = IoGetAttachedDeviceReference(DeviceObject);
+            if (AttachedDisk != DeviceObject) {
+                Status = WnbdGetDiskNumber(
+                    AttachedDisk, (PULONG) &Device->DiskNumber);
+                if (Status) {
+                    WNBD_LOG_WARN("Could not get disk number. Error: %d.",
+                                  Status);
+                }
+            }
+            else {
+                WNBD_LOG_WARN("Couldn't not get disk number. "
+                              "Couldn't get attached PDO.");
+            }
+            ObDereferenceObject(AttachedDisk);
+
+            DWORD RequiredSize = 0;
```

Comment:
Consider moving `RequiredSize` to WnbdGetDiskInstancePath as you did for the type: https://github.com/cloudbase/wnbd/pull/41/files#diff-580945d31a61840d923a5b5a149d0a0e645d44898b80d8eed25ea0713b208e00R468.

Reply:
You mean the ``ReturnedType``? I was thinking about letting the caller adjust the buffer size, if needed.

In this case, we're using WNBD_MAX_NAME_LENGTH(255), which is greater than MAX_DEVICE_ID_LEN(200), so we should be fine.